File: RestaurantHubBackend/src/main/java/com/restauranthub/service/MenuService.java
package com.restauranthub.service;

import java.util.List;
import java.util.Optional;

import com.restauranthub.model.Menu;

public interface MenuService {

    List<Menu> getAllMenuItems();

    Optional<Menu> getMenuById(Long id);

    List<Menu> getMenuByCategory(String category);

    List<Menu> searchFood(String foodName);

    Menu addMenu(Menu menu);

    Menu updateMenu(Long id, Menu menu);

    String deleteMenu(Long id);

}

File: frontend/src/services/menuService.js
import API from "./api";

export const getMenu = () => {

    return API.get("/menu");

};

export const addMenu = (data) => {

    return API.post("/menu", data);

};

export const updateMenu = (id, data) => {

    return API.put(`/menu/${id}`, data);

};

export const deleteMenu = (id) => {

    return API.delete(`/menu/${id}`);

};